### PR TITLE
Handle long URIs

### PR DESCRIPTION
**Pull request by achantavy**
If an escalation policy has over 155 services, this script will create a URL that is too long for urllib3 to handle.

This PR handles that case by splitting up that call into 2 chunks.

### CODE REVIEW

**Review by FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg — COMMENTED** [review]
What's the url length limit? Could we break it up into chunks based on that size and  loop?

**Comment by achantavy** [comment]
> What's the url length limit? Could we break it up into chunks based on that size and loop?

2000 chars I think? I can't find a definitive source.

At this level of the code though we do not have access to the generated string that is ultimately passed to urllib so we can't do much better than handling the resulting exception.

**Review by FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg — DISMISSED** [review]

File: pull_alerts.py [hunk 1]
```diff
@@ -45,11 +46,30 @@ def pretty_output(self):
...
+
+    except urllib.error.HTTPError as e:
+        if e.reason == 'URI Too Long':
+            mid_point = int(len(service_ids)/2)
+            recent_incidents = list(pagerduty_service.incidents.list(
+                service_ids=service_ids[:mid_point],
+                since=datetime.now() - time_window,
+            ))
+            recent_incidents.extend(
+                list(
```
Comment (ikonst) [inline pull_alerts.py]: `lst.extend(list(one_element))` -> `lst.append(one_element)`

File: pull_alerts.py [hunk 2]
```diff
@@ -45,11 +46,30 @@ def pretty_output(self):
...
+                    pagerduty_service.incidents.list(
+                        service_ids=service_ids[mid_point:],
+                        since=datetime.now() - time_window,
+                    )
+                )
+            )
```
Comment (ikonst) [inline pull_alerts.py]:
```suggestion
            mid_point = int(len(service_ids)/2)
            return recent_incidents_for_services(
               service_ids=service_ids[:mid_point],
               time_window=time_window
            ) + recent_incidents_for_services(
               service_ids=service_ids[mid_point:],
               time_window=time_window
            )
```

**Review by ikonst — APPROVED** [review]

**Pull request merged**